File: app/src/main/java/com/soundcloud/android/comments/Comment.java
package com.soundcloud.android.comments;

import com.soundcloud.android.api.legacy.model.PublicApiComment;
import com.soundcloud.android.model.Urn;
import com.soundcloud.android.presentation.ListItem;
import com.soundcloud.java.optional.Optional;

import java.util.Date;

public class Comment implements ListItem {

    private final PublicApiComment apiComment;

    Comment(PublicApiComment apiComment) {
        this.apiComment = apiComment;
    }

    @Override
    public Urn getUrn() {
        return apiComment.getUrn();
    }

    @Override
    public Optional<String> getImageUrlTemplate() {
        return Optional.absent();
    }

    Urn getUserUrn() {
        return apiComment.getUser().getUrn();
    }

    String getUsername() {
        return apiComment.getUser().getUsername();
    }

    long getTimeStamp() {
        return apiComment.timestamp;
    }

    String getText() {
        return apiComment.body;
    }

    Date getDate() {
        return apiComment.getCreatedAt();
    }
}

File: app/src/main/java/com/soundcloud/android/comments/Comment.kt
package com.soundcloud.android.comments;

import com.soundcloud.android.api.legacy.model.PublicApiComment
import com.soundcloud.android.model.Urn
import com.soundcloud.android.presentation.ListItem
import com.soundcloud.java.optional.Optional
import java.util.Date

data class Comment(
        override val urn: Urn,
        val userUrn: Urn,
        val username: String,
        val timeStamp: Long?,
        val text: String,
        val date: Date,
        override val imageUrlTemplate: Optional<String>
) : ListItem {
    constructor(comment: ApiComment) : this(comment.urn, comment.user.urn, comment.user.username, comment.trackTime.orNull(), comment.body, comment.createdAt, comment.user.imageUrlTemplate)
    constructor(comment: PublicApiComment) : this(comment.urn, comment.user.urn, comment.user.username ?: "", comment.timestamp, comment.body, comment.createdAt, comment.user.imageUrlTemplate)
}
